R/load.r

```diff
@@ -148,12 +150,18 @@ load_all <- function(path = ".", reset = TRUE, recompile = FALSE,
     unload_dll(package)
   }
 
-  if (recompile) {
-    pkgbuild::clean_dll(path)
+  # Compile dll if requested
+  if (!missing(recompile) && missing(compile)) {
+    compile <- if (isTRUE(recompile)) TRUE else pkgbuild::needs_compile()
   }
 
-  # Compile dll if it exists
-  pkgbuild::compile_dll(path, quiet = quiet)
+  if (isTRUE(compile)) {
+    pkgbuild::compile_dll(path, force = TRUE, quiet = quiet)
+  } else if (identical(compile, FALSE)) {
```

Comment:
Could we just drop the else conditions entirely here?

Reply:
Yeah, the branching doesn't feel as natural/needed any more.